framework/src/modules/chain/dpos/delegates_info.js

```diff
@@ -162,24 +162,40 @@ class DelegatesInfo {
 		);
 
 		return Promise.all(
-			delegatesWithEarnings.map(({ delegatePublicKey, earnings }) => {
-				const { delegateAccount } = roundSummary.uniqForgersInfo.find(
-					({ publicKey }) => publicKey === delegatePublicKey,
-				);
-
-				const { fee, reward } = earnings;
-				const amount = fee.plus(reward);
-
-				const filters = {
-					publicKey_in: delegateAccount.votedDelegatesPublicKeys,
-				};
-				const field = 'voteWeight';
-				const value = amount.toString();
+			delegatesWithEarnings
+				.filter(({ delegatePublicKey }) => {
```

Comment:
The complexity here for the `.filter` and `.map` functions are too high. 
I suggest to apply some functional programming here where we extract the logic into private functions, example:

```
async _updateVotedDelegatesVoteWeight(roundSummary, undo, tx) {
    const delegatesWithEarnings = this._getDelegatesWithTheirEarnings(
        roundSummary,
    );

    return Promise.all(
        delegatesWithEarnings
            .filter(({ delegatePublicKey }) => _delegatesWithVotesForDelegates)
            .map(({ delegatePublicKey, earnings }) => _updateDelegateVoteWeight),
    );
}
```